src/monosynth.js

```diff
@@ -0,0 +1,358 @@
+   *  @param   {Number} [secondsFromNow] (optional) a time (in seconds
+   *                                     from now) at which
+   *                                     to set the note
+   *  @private
+   */
+  p5.MonoSynth.prototype._setNote = function(note, secondsFromNow) {
+    var freqVal = p5.prototype.midiToFreq(note);
+    this.oscillator.freq(freqVal, 0, secondsFromNow);
+  };
+
+  /**
+     *  Play tells the MonoSynth to start playing a note
+     *  
+     *  @method playNote 
+     *  @param  {Number} [note] midi note to play (ranging from 0 to 127 - 60 being a middle C)
+     *  @param  {Number} [velocity] velocity of the note to play (ranging from 0 to 1)
+     *  @param  {Number} [secondsFromNow]  time from now (in seconds) at which to play
+     *  @param  {Number} [sustainTime] time to sustain before releasing the envelope
+     *  
+     */  
+
+  p5.MonoSynth.prototype.play = function (note, velocity, secondsFromNow, susTime) {
+    // set range of env (TO DO: allow this to be scheduled in advance)
+    var vel = velocity || 1;
+
+    this.triggerAttack(note,velocity,secondsFromNow);
+    this.triggerRelease(secondsFromNow + susTime);
+  };
+
+  /**
+     *  Trigger the Attack, and Decay portion of the Envelope.
+     *  Similar to holding down a key on a piano, but it will
+     *  hold the sustain level until you let go. 
+     *
+     *  @param  {Number} secondsFromNow time from now (in seconds)
+     *  @param  {Number} [velocity] velocity of the note to play (ranging from 0 to 1)
+     *  @param  {Number} [secondsFromNow]  time from now (in seconds) at which to play
+     *  @method  triggerAttack
+     */  
+  p5.MonoSynth.prototype.triggerAttack = function (note, velocity, secondsFromNow) {
+
+    var now = p5sound.audiocontext.currentTime;
+    var tFromNow = secondsFromNow || 0;
+    var t = now + tFromNow;
+    var n = p5.prototype.midiToFreq(note);
+    
+    this._isOn = true;
+    this.oscillator.freq(n, 0, t);
+    this.env.ramp(this.output, t, velocity);
+  };
+
+  /**
+     *  Trigger the Release of the Envelope. This is similar to releasing
+     *  the key on a piano and letting the sound fade according to the
+     *  release level and release time.
+     *  
+     *  @param  {Number} secondsFromNow time to trigger the release
+     *  @method  triggerRelease
+     */  
+
+  p5.MonoSynth.prototype.triggerRelease = function (secondsFromNow) {
+    this.env.ramp(this.output, secondsFromNow, 0);
+    this._isOn = false;
+  };
+
+
+  /**
+     *  Set cutoms parameters to a specific synth implementation.
+     *  This method does nothing by default unless you implement
+     *  something for it.
+     *  For instance if you want to build a complex synthesizer
+     *  with one or more filters, effects etc. this is where you
+     *  will want to set their values.
+     *  
+     *  @method  setParams
+     *  @param   
+     * 
+     */  
+
+  p5.MonoSynth.prototype.setParams = function(params) {
```

Comment:
let's leave this method out since it does nothing, it'll be distracting for most users. We could instead include an example for how to extend the MonoSynth